administrator/components/com_content/controllers/article.php

```diff
@@ -145,6 +146,31 @@ public function batch($model = null)
 	}
 
 	/**
+	 * Method to generate and store share token.
+	 *
+	 * @return  boolean   True if token successfully stored, false otherwise and internal error is set.
+	 *
+	 * @since   3.7
+	 */
+	public function shareDraft()
+	{
+		$title = $this->input->getArray(array('jform' => array('title' => 'string')));
+
+		// Get the model
+		$model = $this->getModel();
+		$return = $model->shareToken($title);
+
+		if ($return)
+		{
+			JText::_('COM_CONTENT_TOKEN_SAVED');
```

Comment:
@nikitadhiman The AJAX call will be initiated from the view by clicking on a button.